extensions-contrib/opentelemetry-extensions/src/test/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReaderTest.java

```diff
@@ -0,0 +1,280 @@
+            .addAttributesBuilder() // test sum with attributes
+            .setKey(METRIC_ATTRIBUTE_COLOR)
+            .setValue(AnyValue.newBuilder().setStringValue(METRIC_ATTRIBUTE_VALUE_RED).build());
+
+    // Create Second Metric
+    Metric.Builder gaugeMetricBuilder = metricsDataBuilder.addResourceMetricsBuilder()
+            .addInstrumentationLibraryMetricsBuilder()
+            .addMetricsBuilder();
+
+    metricsDataBuilder.getResourceMetricsBuilder(1)
+            .getResourceBuilder()
+            .addAttributes(KeyValue.newBuilder()
+                            .setKey(RESOURCE_ATTRIBUTE_ENV)
+                            .setValue(AnyValue.newBuilder().setStringValue(RESOURCE_ATTRIBUTE_VALUE_DEVEL))
+                            .build());
+
+    metricsDataBuilder.getResourceMetricsBuilder(1)
+            .getInstrumentationLibraryMetricsBuilder(0)
+            .getInstrumentationLibraryBuilder()
+            .setName(INSTRUMENTATION_LIBRARY_NAME)
+            .setVersion(INSTRUMENTATION_LIBRARY_VERSION);
+
+    gaugeMetricBuilder.setName("example_gauge")
+            .getGaugeBuilder()
+            .addDataPointsBuilder()
+            .setAsInt(8)
+            .setTimeUnixNano(TIMESTAMP)
+            .addAttributesBuilder() // test sum with attributes
+            .setKey(METRIC_ATTRIBUTE_FOO_KEY)
+            .setValue(AnyValue.newBuilder().setStringValue(METRIC_ATTRIBUTE_FOO_VAL).build());
+
+    MetricsData metricsData = metricsDataBuilder.build();
+
+    CloseableIterator<InputRow> rows = new OpenTelemetryMetricsProtobufReader(
+            dimensionsSpec,
+            new ByteEntity(metricsData.toByteArray()),
+            "metric.name",
+            "raw.value",
+            "descriptor.",
+            "custom."
+    ).read();
+
+    Assert.assertTrue(rows.hasNext());
+    InputRow row = rows.next();
+
+    Assert.assertEquals(4, row.getDimensions().size());
+    assertDimensionEquals(row, "metric.name", "example_sum");
+    assertDimensionEquals(row, "custom.country", "usa");
+    assertDimensionEquals(row, "descriptor.color", "red");
+    assertDimensionEquals(row, "raw.value", "6");
+
+    Assert.assertTrue(rows.hasNext());
+    row = rows.next();
+    Assert.assertEquals(4, row.getDimensions().size());
+    assertDimensionEquals(row, "metric.name", "example_gauge");
+    assertDimensionEquals(row, "custom.env", "devel");
+    assertDimensionEquals(row, "descriptor.foo_key", "foo_value");
+    assertDimensionEquals(row, "raw.value", "8");
+
+  }
+
+  @Test
+  public void testDimensionSpecExclusions()
+  {
+    metricBuilder.setName("example_gauge")
+            .getGaugeBuilder()
+            .addDataPointsBuilder()
+            .setAsInt(6)
+            .setTimeUnixNano(TIMESTAMP)
+            .addAttributesBuilder() // test sum with attributes
+            .setKey(METRIC_ATTRIBUTE_COLOR)
+            .setValue(AnyValue.newBuilder().setStringValue(METRIC_ATTRIBUTE_VALUE_RED).build());
```

Comment:
we should add some more attributes to ensure we still preserve resource and metric attributes that are present in the data. Right now we're only testing that all the attributes are gone, but the lack of dimension spec should still allow pass-through of attributes that are not listed in the exclusion.

Reply:
Makes sense. I added that case.